### DISCUSSION TITLE
Problems with NFC and Libre 2

### DISCUSSION THREAD

**Discussion by Rabuffulin** [NightscoutFoundation/xDrip#2958] [General]
Hi...I'm trying since yesterday start a new Libre2 sensor on my Samsung ZFold 4 but it's not working. Xdrip send me messages NFC error IO or NFC failed. Last month I had this problem and I lost 2 sensors because this, but when I put third sensor, it works. Anybody already had this problem? I'm two days without sensor because all sensor that I put I have the same issue....I try with 3 sensors and the problem is the same....The sensor appears to me on Bluetooth map but when I choose it Xdrip can't connect and I don't know what to do...could help me?

**Comment by DayZer035** [comment]
Hi

Sorry I can't help but I've also had similar problems for the first time ever yesterday.

New sensor yesterday. Started with Librelink on Samsung S21 Ultra. When trying to start on Xdrip via NFC scan I now just get 'NFC read timeout' error. I can't get any connection or readings at all, let alone bluetooth. I've even tried this with LL uninstalled.

Prior to this I had been using Libre 2 bluetooth for 6 months without problem, Since the LL update I was also able to run both LL and Xdrip in parallel with both receiving readings via bluetooth.

I haven't yet tried a new sensor, I wanted to exhaust other ideas first!

My only workaround is to use LibreLinkUp to get values into xdrip via 'Web Follower' setting. However, this is a poor alternative as I can't calibrate, there is a time lag and it requires an internet connection.

**Comment by DayZer035** [comment]
I fixed following these instructions:

https://github.com/NightscoutFoundation/xDrip/discussions/2845#discussioncomment-5995246

**Reply by Rabuffulin** [reply]
Yes, I tried this option and works for me....tks.

**Comment by PhilTU** [comment]
OK, so it's a sensor version error requiring update of xDrip+ to nightly version 25/5/23 and OOP2 to same.

NFC scan was ok, but the data could not be processed due to change of sensor information by the vendor.

**Comment by Koral00920** [comment]
Hi!

I have a problem with my libre 2 sensor connecting to xdrip. It's been almost a year since I've started using libre 2 with xdrip and oop2 and everything was fine until yesterday. Yesterday I changed my sensor as always (did not make any changes in settings) but since then I cannot scan it with xdrip to make a connection. First I get the NFC timeout message, then it finally says scanned OK but shortly after that I get the 'xdrip needs oop2 to handle libre2/US sensors' error... I have newest versions of both xdrip and oop2 installed and everything worked till yesterday! I reinstalled xdrip and oop2, deleted all my settings and tried to set it up again from scratch. Nothing works. My bg is getting high because I can't access constant readings and my aaps can't work because of no glucose data. Please, help!

**Reply by Navid200** [reply]
@psonnera

**Reply by PhilTU** [reply]
Is OOP2 from late May this year? (05/23)

**Reply by Koral00920** [reply]
The app says it's from 2022.06, so no. But I can't find the newer version, can somebody send a link? 🙏

**Reply by FuLiDu** [reply]
@Koral00920 https://xdrip.readthedocs.io/en/latest/use/OOP/

**Comment by psonnera** [comment]
Can you share a few screenshots of the events log after you scan with NFC?

**Reply by Koral00920** [reply]
![Screenshot_20231129-155533_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/39039d50-a414-4c95-81a7-25bdd3f2e3c9)
![Screenshot_20231129-155526_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/449bc065-8509-4097-94ae-a9ddefaf9d4b)
![Screenshot_20231129-155520_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/8336db88-178b-4587-a163-92b2c07486ef)
![Screenshot_20231129-155513_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/3e8c83c0-8737-4450-a300-af77c2a0b772)
![Screenshot_20231129-155502_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/85f57e9c-67e1-4432-9091-370f37f217e9)
![Screenshot_20231129-155457_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/fee3e60e-154a-4345-9154-6219a8a42117)
![Screenshot_20231129-155507_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/e3eb44e6-012b-45e2-9b83-a84da68e1cf7)

**Comment by Koral00920** [comment]
![Screenshot_20231129-121844_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/e200730e-de10-444d-82d0-4c4054ea064b)
![Screenshot_20231129-121831_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/dc9db2f5-4492-46b9-a560-b00cafc23d70)
![Screenshot_20231129-121823_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/22936ad4-082a-4eaa-b065-4e2561cda8c7)
![Screenshot_20231129-121812_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/56688590/35ed7685-835a-45db-8e36-c30e36a31492)

**Reply by Koral00920** [reply]
Thanks for replying! Here are the logs after I scanned with xdrip, it said scanned ok. It looks like it can't connect to oop2 though it is running.

**Reply by psonnera** [reply]
Looks like OOP2 is not answering, can you post a screenshot of the OOP2 app?

**Reply by Koral00920** [reply]
![Screenshot_20231129-134335_OOPalgorithm2](https://github.com/NightscoutFoundation/xDrip/assets/56688590/8dcfed57-8b70-47e3-8c0d-067628313140)

**Reply by psonnera** [reply]
Update OOP2 https://xdrip.readthedocs.io/en/latest/use/OOP/#oop2

**Reply by Koral00920** [reply]
It's working! Thank you so much! I never got to this site and when I was searching for oop2 in different places I kept downloading what was linked as the latest update. I didn't realize it was out of date too. It's only weird that it stopped working so suddenly...

Anyway, many thanks again!

**Comment by PhilTU** [comment]
> I kept downloading what was linked as the latest update. I didn't realize it was out of date too

the internet has a long memory, no version or quality control. The change would be in the sensor, needed new OOP2 for revised sensor type.

**Reply by Koral00920** [reply]
Now it makes sense. Thanks!

**Reply by avicorp** [reply]
Hey, I experiencing the issue and follow the guide on updating and configuring xdrip.

Can you help please 🥺?
![Screenshot_20240304_091520_OOPalgorithm2](https://github.com/NightscoutFoundation/xDrip/assets/7301959/d0a3b7c8-6737-4641-a672-a7d2fa7268d6)

![Screenshot_20240304_091537_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/7301959/018b8ebc-6e80-4478-8d5d-cc363757d9f5)

![Screenshot_20240304_091735_Settings](https://github.com/NightscoutFoundation/xDrip/assets/7301959/1a1a63c1-d23f-4e1b-a2f1-a1c5112f8578)

**Reply by psonnera** [reply]
Enable also foreground and disable OOP1

**Reply by avicorp** [reply]
Workd!!

**Comment by eboocha** [comment]
Hi, in Malaysia here, Abbott just brought in Libre2 plus. Before this I'm using Libre1 with MiaoMiao3 and XDrip+. So now the new Libre2 not supported by MiaoMiao3. LibreLink also not available in Malaysia here. So I only can use Reader.
Then I followed this:
https://androidaps.readthedocs.io/he/latest/CompatibleCgms/Libre2MinimalL00per.html#minimallooper-definitions
But when I NFC scan the sensor with my phone, it keeps showing "NFC read timeout" and sometimes "NFC IO error"

XDrip+ version 041244d-2025.05.09
OOP2 version 27da3f5-2025.06.12
![17585489379843862059054063856569](https://github.com/user-attachments/assets/acc8d1c9-e8cf-4d02-a0b1-5385b17a18dc)
![17585489595588032175422531489377](https://github.com/user-attachments/assets/41175548-392b-4a60-8633-03352bf108d3)

Can somebody please help me? Because my daughter using this sensor and I very rely XDrip to monitor her BG

**Reply by PhilTU** [reply]
Update xDrip to nightly 20/6/25 or later.

https://github.com/NightscoutFoundation/xDrip/discussions/4176#discussion-8918766

**Reply by eboocha** [reply]
Thanks for your reply.
I just updated:
XDrip+ 1490f37-2025.06.20
OOP2 27da3f5-2025.06.12
But still "NFC read timeout"

Then I updated XDrip+ version d72d03c-2025.09.19
Still cannot connect.

By the way, my Libre2+ SN starting 3MH instead of 301
![17586346714546184932196147305485](https://github.com/user-attachments/assets/f7b181db-e13a-4a8e-bad7-8796c1faa97f)

**Reply by PhilTU** [reply]
Screenshot with NFC reader section would be handy.

Is "use faster multiblock method" disabled in xDrip settings?

**Reply by eboocha** [reply]
![175864806156382937556383295928](https://github.com/user-attachments/assets/a84f1b38-684f-412a-81ea-b768e7413c0a)
![1758648118072691396055182942234](https://github.com/user-attachments/assets/1b0bbd27-c428-4358-a29f-370173dcc89c)
![17586481517342460348917878963055](https://github.com/user-attachments/assets/1f68fb7a-aad6-44c8-8c79-62b4c4e837a7)
![17586481642274813707678352302123](https://github.com/user-attachments/assets/823ecc55-a56c-4bc5-b977-45b9e1f25eb9)
![17586482409354975038106282553906](https://github.com/user-attachments/assets/9ab77e3a-dedb-4fca-8298-ed42aa42a994)

**Reply by PhilTU** [reply]
I always have the last NFC option enabled.
<img width="1200" height="2670" alt="Screenshot_20250923-212538 Chrome" src="https://github.com/user-attachments/assets/30007b00-8b00-458b-8020-a8136f2de588" />

**Reply by PhilTU** [reply]
If you search error logging for NFC it might show something

**Reply by PhilTU** [reply]
<img width="1200" height="2670" alt="Screenshot_20250923-212858 xDrip_Test" src="https://github.com/user-attachments/assets/1d10cf2e-6368-4db2-8aab-23a374f0c1f9" />

**Reply by eboocha** [reply]
![17586696709943695037309300245442](https://github.com/user-attachments/assets/35e6654a-e97c-4198-a2c3-9fe98c9c734c)

Why Libre1 detected??

![17586697302116219028137012604146](https://github.com/user-attachments/assets/d873cb12-2d18-43f4-b54b-536f78b2c4c7)
![17586697492507121028322741748067](https://github.com/user-attachments/assets/92117433-2b9b-49bf-a4d3-6cd8985bde1a)

**Reply by PhilTU** [reply]
See the Issues section re new sensor in Poland. Unrecognised sensor so falls back to Libre 1.

Try the test version from the issues for a start.

**Reply by PhilTU** [reply]
> Why Libre1 detected??

go to setting->more settings ->extra tags for logging and add there NFCReaderX:v
After that please scan again, you should see in the log something like:
'' patchInfo = 0xdddddd"

Please upload this picture to this thread. This will show sensor type code.

**Reply by eboocha** [reply]
![17587186139965964562490699473048](https://github.com/user-attachments/assets/feb40615-ca97-49e7-96db-3e15f16b92c7)
![17587186284855775783107638269421](https://github.com/user-attachments/assets/e65e9037-7018-4f0d-8443-235bb94c06d8)

**Reply by PhilTU** [reply]
I think this is sensor type same as Brazil. xDrip+ does not currently have support for this.

<img width="369" height="58" alt="image" src="https://github.com/user-attachments/assets/230abe58-3f7e-420b-a4fd-43ca899076be" />

**Reply by eboocha** [reply]
Oh no... Hopefully soon will have update for this version. Can't live without XDrip.
Thank you so much for your assist. Appreciate it 👍

**Reply by mhalano** [reply]
Oh! I have this sensor, I live in Brazil too. I also tried using Jugglerco, which works well and without using OOP2, and also communicates with my Pixel Watch 3 running Wear OS 6, but the interface is quite confusing.

**Reply by PhilTU** [reply]
No progress I'm afraid.
